packages/data-context/src/sources/migration/codegen.ts

```diff
@@ -340,91 +340,86 @@ export function renameSupportFilePath (relative: string) {
 }
 
 export function reduceConfig (cfg: LegacyCypressConfigJson): ConfigOptions {
-  const excludedFields = ['pluginsFile', '$schema']
-
   return Object.entries(cfg).reduce((acc, [key, val]) => {
-    if (excludedFields.includes(key)) {
-      return acc
-    }
-
-    if (key === 'e2e' || key === 'component') {
-      const value = val as ResolvedConfigOptions
-
-      if (!value) {
+    switch (key) {
+      case 'pluginsFile':
+      case '$schema':
         return acc
-      }
 
-      const { testFiles, ignoreTestFiles, ...rest } = value
+      case 'e2e':
+      case 'component': {
+        const value = val as ResolvedConfigOptions
 
-      // don't include if it's the default! No need.
-      const specPattern = getSpecPattern(cfg, key)
-      const ext = '**/*.cy.{js,jsx,ts,tsx}'
-      const isDefaultE2E = key === 'e2e' && specPattern === `cypress/e2e/${ext}`
-      const isDefaultCT = key === 'component' && specPattern === ext
+        if (!value) {
+          return acc
+        }
+
+        const { testFiles, ignoreTestFiles, ...rest } = value
+
+        // don't include if it's the default! No need.
+        const specPattern = getSpecPattern(cfg, key)
+        const ext = '**/*.cy.{js,jsx,ts,tsx}'
+        const isDefaultE2E = key === 'e2e' && specPattern === `cypress/e2e/${ext}`
+        const isDefaultCT = key === 'component' && specPattern === ext
```

Comment:
not a change for this PR, just in general..would love to see these default values pulled from the config packages instead of hard-coded here